drivers/wifi/siwx91x/siwx91x_wifi.c

```diff
@@ -295,6 +295,96 @@ static int siwx91x_ap_disable(const struct device *dev)
 	return ret;
 }
 
+static bool siwx91x_param_comparison(struct wifi_iface_status *prev_params,
```

Comment:
I suggest to try to change the name of the function in a way where the meaning of the returned value is explicit. eg. `siwx91x_param_changed()`